ifrs17/Test/QueriesScenarioDataTest.ipynb

```diff
@@ -0,0 +1,386 @@
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "queriedRawVars.SelectMany(x => x.Values).Sum().Should().Be(0);"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## Scenario query, Workspace with no new scenario data and DataSource with scenario data"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "await DataSource.DeleteAsync( await DataSource.Query<RawVariable>().ToArrayAsync() );",
+                "\nawait DataSource.UpdateAsync( bestEstimateRawVars.Concat(scenarioRawVars).ToArray() );",
+                "\nvar ws = Workspace.CreateNew();",
+                "\nws.Initialize(x => x.FromSource(DataSource).DisableInitialization<RawVariable>().DisableInitialization<IfrsVariable>());"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "var queriedRawVars = await ws.QueryPartitionedDataAsync<RawVariable,PartitionByReportingNodeAndPeriod>(DataSource, partitionScenarioMTUP.Id, partition.Id, ImportFormats.Cashflow);"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "queriedRawVars.SelectMany(x => x.Values).Sum().Should().Be(3.2);"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## Scenario query, Workspace with no new scenario data and DataSource with only Best Estimate data"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "await DataSource.DeleteAsync( await DataSource.Query<RawVariable>().ToArrayAsync() );",
+                "\nawait DataSource.UpdateAsync( bestEstimateRawVars.ToArray() );",
+                "\nvar ws = Workspace.CreateNew();",
+                "\nws.Initialize(x => x.FromSource(DataSource).DisableInitialization<RawVariable>().DisableInitialization<IfrsVariable>());"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "var queriedRawVars = await ws.QueryPartitionedDataAsync<RawVariable,PartitionByReportingNodeAndPeriod>(DataSource, partitionScenarioMTUP.Id, partition.Id, ImportFormats.Cashflow);"
```

Comment:
QueryPartitionedDataAsync behaves differently if the type is RawVariable or IfrsVariable. 
I would expect tests with both types.

Reply:
done